build/cmake/Fuzzer.cmake

```diff
@@ -0,0 +1,14 @@
+include (CheckCSourceCompiles)
+include (CMakePushCheckState)
+include (MongoSettings)
+
+mongo_setting(
+  MONGO_FUZZ "Whether fuzz testing is enabled"
+  TYPE BOOL
+  DEFAULT FALSE
+)
+
+if (MONGO_FUZZ)
+set(CMAKE_C_FLAGS "-fsanitize=undefined -fsanitize=address -fsanitize=fuzzer-no-link" CACHE STRING "Custom comp" FORCE)
+set(MONGO_FUZZ_ENGINE -fsanitize=fuzzer)
+endif ()
```

Comment:
```suggestion
```
Remove in favor of reusing the existing `MONGO_SANITIZE` option. Add the following after the `MONGO_SANITIZE` option definition in `build/cmake/Sanitizers.cmake`:

```cmake
if (MONGO_FUZZ)
    set(mongo_fuzz_options "address,undefined,fuzzer-no-link")

    if (NOT "${MONGO_SANITIZE}" STREQUAL "${mongo_fuzz_options}")
        message(WARNING "Overriding user-provided MONGO_SANITIZE options in favor of MONGO_SANITIZE=ON")
    endif()

    set_property (CACHE MONGO_SANITIZE PROPERTY VALUE "${mongo_fuzz_options}")
endif ()
```

Reply:
Done